agent-discoveries-backend/src/routes/user.ts

```diff
@@ -46,6 +46,73 @@ function createRouter(db: Knex) {
     }
   })
 
+  router.put('/update', auth.isLoggedIn, async (req, res) => {
+    const { password, imageUrl } = req.body
+
+    interface Changes {
+      hashedPassword?: string
+      imageUrl?: string
+    }
+
+    const changes: Changes = {}
+
+    if (password !== undefined) {
+      const hashedPassword = await hashPassword(password)
+      changes.hashedPassword = hashedPassword
+    }
+    if (imageUrl !== undefined) {
+      changes.imageUrl = imageUrl
+    }
+
+    try {
+      await db('users')
+        .where({ userId: req.session.user?.userId })
+        .update(changes)
+      res.status(200).send()
+    } catch (e) {
+      res.status(400).send()
+    }
+  })
+
+  router.put('/delete', auth.isLoggedIn, async (req, res) => {
+    const { user } = req.body
+    if (user === undefined) {
+      return res.status(400).send('must specify a user to delete')
+    }
+
+    if (req.session.user?.username !== user) {
+      // if we are trying to delete another user
+      let dbResponse = await db('users')
+        .select('admin')
+        .where({ userId: req.session.user?.userId })
+
+      let admin = dbResponse[0].admin
+      if (!admin) {
+        return res.status(403).send('non-admin cannot delete other users')
+      }
+    }
+
+    try {
+      await db('users').delete().where({ username: user })
+      res.status(200).send()
+    } catch (e) {
+      res.status(400).send()
+    }
+  })
+
+  router.get('/get', auth.isLoggedIn, async (req, res) => {
```

Comment:
Ah, so for interacting with endpoints like these in a RESTful way - you want to do operations on specific users. E.g. for this `users/get` endpoint, you'll want to define the path as `'/:userId'` or similar, so you can GET a `{baseURL}/user/1` (user with id 1)

Reply:
Changes implemented.
Can no longer request a list of ALL users in one go